common/djangoapps/student/views.py

```diff
@@ -1804,6 +1805,27 @@ def create_account_with_params(request, params):
     return new_user
 
 
+def _enroll_user_in_pending_courses(student):
+    """
+    Enroll student in any pending courses he/she may have.
+    """
+    if student:
```

Comment:
great! Now there's no reason to do this check here!

or — it might read cleaner this way — if we're feeling very paranoid, we can first check if there's no student, log an error if there isn't, and then do the rest of the function:

```python
if not student:
    log.error('no student blah blah blah....')
    return

<...>
```

Also, it looks like this function is only called for its side effects, so there's no reason to return anything at the end.